aws/resource_aws_config_remediation_configuration_test.go

```diff
@@ -0,0 +1,324 @@
+			},
+
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckConfigRemediationConfigurationName(n, desired string, obj *configservice.RemediationConfiguration) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.Attributes["cnofig_rule_name"] != *obj.ConfigRuleName {
+			return fmt.Errorf("Expected name: %q, given: %q", desired, *obj.ConfigRuleName)
+		}
+		return nil
+	}
+}
+
+func testAccCheckConfigRemediationConfigurationExists(n string, obj *configservice.RemediationConfiguration) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No config rule ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).configconn
+		out, err := conn.DescribeRemediationConfigurations(&configservice.DescribeRemediationConfigurationsInput{
+			ConfigRuleNames: []*string{aws.String(rs.Primary.Attributes["name"])},
+		})
+		if err != nil {
+			return fmt.Errorf("Failed to describe config rule: %s", err)
+		}
+		if len(out.RemediationConfigurations) < 1 {
+			return fmt.Errorf("No config rule found when describing %q", rs.Primary.Attributes["name"])
+		}
+
+		rc := out.RemediationConfigurations[0]
+		*obj = *rc
+
+		return nil
+	}
+}
+
+func testAccCheckConfigRemediationConfigurationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).configconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_config_remediation_configuration" {
+			continue
+		}
+
+		resp, err := conn.DescribeRemediationConfigurations(&configservice.DescribeRemediationConfigurationsInput{
+			ConfigRuleNames: []*string{aws.String(rs.Primary.Attributes["name"])},
+		})
+
+		if err == nil {
+			if len(resp.RemediationConfigurations) != 0 &&
+				*resp.RemediationConfigurations[0].ConfigRuleName == rs.Primary.Attributes["name"] {
+				return fmt.Errorf("remediation configuration(s) still exist for rule: %s", rs.Primary.Attributes["name"])
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccConfigRemediationConfigurationConfig_basic(randInt int) string {
+	return fmt.Sprintf(`
+resource "aws_config_remediation_configuration" "foo" {
+	config_rule_name = "${aws_config_config_rule.foo.name}"
+
+	resource_type = ""
+	target_id = "SSM_DOCUMENT"
+	target_type = "AWS-PublishSNSNotification"
+	target_version = "1"
+
+	parameters = [
+		{
+			resource_value = "Message"
+		},
+		{
+			static_value = {
+				key = "TopicArn"
+				value = "${aws_sns_topic.foo.arn}"
+			}
+		},
+		{
+			static_value = {
+				key = "AutomationAssumeRole"
+				value = "${aws_iam_role.aar.arn}"
+			}
+		}
+	]
+
+	depends_on = [
```

Comment:
Nit: The Terraform configuration contains implicit references to these resources (e.g. `value = "${aws_sns_topic.foo.arn}"`) so `depends_on` is extraneous. 👍